File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        if(n == 1 || n == 2 || n == 3) return n;
        int[] answer = {1, 2, 3};
        
        for(int i = 0; i < n-3; i++){
            answer[0] = answer[1];
            answer[1] = answer[2];
            answer[2] = answer[0] + answer[1];
        }
        return answer[2];
        
    }
}

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n < 2:
            return 1
        stairs = [0 for _ in range(n+1)]

        stairs[1] = 1
        stairs[2] = 2

        for i in range(3, n+1):
            stairs[i] = stairs[i-1] + stairs[i-2]

        return stairs[n]
        